### PR TITLE
Added update-all.sh and verify-all.sh to simplify API update process

### PR DESCRIPTION
**Pull request by soltysh**
Based on upstream's scripts. @smarterclayton ptal

### CODE REVIEW

**Comment by soltysh** [comment]
@stevekuznetsov ptal as well, since you're the bash-master here :wink:

File: hack/update-all.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	generated-conversions
+	generated-deep-copies
+	generated-docs
+	generated-swagger-descriptions
+	generated-swagger-spec"
+
+
+for t in $BASH_TARGETS
+do
+	echo -e "${color_yellow}Updating $t${color_norm}"
```
Comment (stevekuznetsov) [inline hack/update-all.sh]: We have `hack/text.sh` library for this.

File: hack/update-all.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	if $SILENT ; then
+		if ! bash "$OS_ROOT/hack/update-$t.sh" 1> /dev/null; then
```
Comment (stevekuznetsov) [inline hack/update-all.sh]: Technically someone could put something other than Bash in their shebang, so we don't really need the `bash` her

File: hack/verify-all.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  for t in $(ls ${pattern})
+  do
+    if is-excluded "${t}" ; then
+      echo "Skipping ${t}"
+      continue
+    fi
+    echo -e "Verifying ${t}"
+    local start=$(date +%s)
+    run-cmd bash "${t}" && tr=$? || tr=$?
```
Comment (stevekuznetsov) [inline hack/verify-all.sh]: Why not use the native `if` like above?

File: hack/verify-all.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    if [[ ${tr} -eq 0 ]]; then
+      echo -e "${color_green}SUCCESS${color_norm}  ${t}\t${elapsed}s"
+    else
+      echo -e "${color_red}FAILED${color_norm}   ${t}\t${elapsed}s"
+      ret=1
+    fi
+  done
+}
+
+while getopts ":v" opt; do
```
Comment (stevekuznetsov) [inline hack/verify-all.sh]: We don't have any precedents for this, verbosity in our scripts is always `VERBOSE= true`.

File: hack/verify-all.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  esac
+done
+
+if ${SILENT} ; then
+  echo "Running in the silent mode, run with -v if you want to see script logs."
+fi
+
+ret=0
+run-checks "${OS_ROOT}/hack/verify-*.sh"
+exit ${ret}
```
Comment (stevekuznetsov) [inline hack/verify-all.sh]: Not a fan of implicitly setting `$ret`

File: hack/update-all.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+BASH_TARGETS="
+	generated-completions
+	generated-conversions
+	generated-deep-copies
+	generated-docs
+	generated-swagger-descriptions
+	generated-swagger-spec"
+
+
+for t in $BASH_TARGETS
```
Comment (stevekuznetsov) [inline hack/update-all.sh]: Prefer `for ...; do`

File: hack/verify-all.sh [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+declare -r color_norm="\033[0m"
+
+# Excluded checks are always skipped.
+EXCLUDED_CHECKS=(
+  "verify-golint.sh"
+  "verify-jsonformat.sh"
+  "verify-open-ports.sh"
+)
+
+function is-excluded {
```
Comment (stevekuznetsov) [inline hack/verify-all.sh]: Prefer `function ...() {`

**Comment by stevekuznetsov** [comment]
In general, a lot of the return value checking / coloring could be done by `os::cmd::expect_success`, I guess

**Comment by soltysh** [comment]
@stevekuznetsov thx, will rework.

**Comment by smarterclayton** [comment]
Since we have make verify, why do we need verify-all.sh? Why not just add
make update?

I'd rather have make contain a list of scripts, than make these scripts.

**Comment by soltysh** [comment]
That's fair. I'll change to that, in that case.

**Comment by soltysh** [comment]
But I'd like to have a separate make target that will run just the scripts, if you don't mind.

**Comment by soltysh** [comment]
I mean for verify.

**Comment by smarterclayton** [comment]
Make lets you skip targets - you can easily skip bill

**Comment by smarterclayton** [comment]
Build

**Comment by smarterclayton** [comment]
make verify -o build

**Comment by soltysh** [comment]
@smarterclayton better?
 
@mfojtik you were asking about it last time

**Comment by stevekuznetsov** [comment]
I like this `make` target... and I like it a lot that it will build for me first. I hate updating without building because I'm dumb and forgot ...

**Comment by smarterclayton** [comment]
LGTM [test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**